tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -66,11 +69,74 @@ static int64_t multiply_dims(llvm::ArrayRef<int64_t> dims, int64_t res = 1) {
   return res;
 }
 
-static int64_t count_dynamic_dims(llvm::ArrayRef<int64_t> dims) {
-  int64_t count = 0;
-  for (auto dim : dims)
-    if (ShapedType::isDynamic(dim)) ++count;
-  return count;
+static OpFoldResult multiply_dims(ArrayRef<OpFoldResult> dims,
+                                  OpBuilder& builder, Location loc,
```

Comment:
Look into making an `ImplicitLocOpBuilder`. It allows you to pass an `OpBuilder` around that matches `Location` correctly.

Reply:
It is not obvious to me that `ImplicitLocOpBuilder` preserves all desired behaviors, since it is a subclass of `OpBuilder` and these builders are typically starting from a `PatternRewriter`. I think using `ImplicitLocOpBuilder` may lose some of the logging mechanisms associated with the rewriter classes.